### PR TITLE
[WebAssembly] Misc. refactoring in AsmTypeCheck (NFC)

### PR DESCRIPTION
**Pull request by aheejin**
Existing methods in AsmTypeCheck assumes symbol operand is the 0th operand; they take a `MCInst` and take `getOperand(0)` on it. I think passing a `MCOperand` removes this assumption and also is more intuitive. This was motivated by a new `try_table` instruction, whose support is going to be added to AsmTypeCheck soon, which has tag symbol operands in any position, depending on the number and the kinds of catch clauses. This PR changes all methods' signature that assumes the 0th operand is the relevant one, even if it's not the symbol operand.

This also adds `getSignature` method, which factors out the common task when getting a `WasmSignature` from a `MCOperand`.

### CODE REVIEW

File: llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmTypeCheck.cpp [hunk 1]
```diff
@@ -112,9 +112,9 @@ bool WebAssemblyAsmTypeCheck::popRefType(SMLoc ErrorLoc) {
   return false;
 }
 
-bool WebAssemblyAsmTypeCheck::getLocal(SMLoc ErrorLoc, const MCInst &Inst,
+bool WebAssemblyAsmTypeCheck::getLocal(SMLoc ErrorLoc, const MCOperand &Op,
```
Comment (dschuff) [inline llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmTypeCheck.cpp:115]: Would it make sense to give these parameters a more-descriptive name, e.g. `SymbolOp` or something?

**Review by dschuff — APPROVED** [review]
this is a nice improvement.

File: llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmTypeCheck.cpp (hunk 1 above)
Comment (aheejin) [reply llvm/lib/Target/WebAssembly/AsmParser/WebAssemblyAsmTypeCheck.cpp]: Changed `Op` to `SymOp`(in `getSymRef`), `LocalOp` (in `getLocal`), `GlobalOp` (in `getGlobal`), and `SigOp` (in `getSignature`).

**Comment by aheejin** [comment]
@llvm-ci That failure doesn't seem to be related to this PR. That's a flang failure.

**Pull request merged**